### PR TITLE
PrettierLintBear.py: Fix syntax error regex

### PR DESCRIPTION
**Pull request by anctartica**
Fixes https://github.com/coala/coala-bears/issues/2365

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: bears/js/PrettierLintBear.py [hunk 1]
```diff
@@ -23,13 +23,17 @@ class PrettierLintBear:
                   'Multi-line objects'}
     SEE_MORE = 'https://prettier.io/'
 
-    regex = re.compile(r'L(?P<line>\d+)C(?P<column>\d+): (?P<message>.*)')
```
Comment (jayvdb) [inline bears/js/PrettierLintBear.py]: im wondering how this could have been so wrong?

Comment (jayvdb) [reply bears/js/PrettierLintBear.py]: resolved

File: bears/js/PrettierLintBear.py [hunk 2]
```diff
@@ -23,13 +23,17 @@ class PrettierLintBear:
...
+    regex = re.compile(
+        r'(?P<message>SyntaxError:.*) '
+        r'\((?P<line>\d+):(?P<column>\d+)\)')
 
     def process_output(self, output, filename, file):
         stdout, stderr = output
-        yield from self.process_output_corrected(stdout, filename, file)
-        yield from self.process_output_regex(stderr, filename, file,
-                                             self.regex)
+        if stdout == '\n':
```
Comment (jayvdb) [inline bears/js/PrettierLintBear.py]: why is this `if` needed?

Comment (anctartica) [reply bears/js/PrettierLintBear.py]: When there is a syntax error stdout has a `\n`. So I suppose we should only show the syntax error in this case.

Comment (jayvdb) [reply bears/js/PrettierLintBear.py]: But the regex being applied to `stdout` will ignore the `\n`..?

Comment (anctartica) [reply bears/js/PrettierLintBear.py]: no we are using `process_output_corrected` since the bear prints corrected output to `stdout`

Comment (jayvdb) [reply bears/js/PrettierLintBear.py]: That feels like a bug in `process_output_corrected`.  If there was an error, the linter can only emit `\n` on stdout.  It shouldnt decide that means the file should be blanked.

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
@gitmate-bot ff

File: bears/js/PrettierLintBear.py (hunk 2 above)
Comment (jayvdb) [reply bears/js/PrettierLintBear.py]: fwiw https://github.com/coala/coala/issues/5415

**Pull request merged**